ekncontent/ekncontent/eknc-xapian-bridge.c

```diff
@@ -330,6 +350,72 @@ send_json_ld_request (EkncXapianBridge *self,
 }
 
 /**
+ * eknc_xapian_bridge_test:
+ * @self: the xapian bridge object
+ * @cancellable: (allow-none): optional #GCancellable object, %NULL to ignore.
+ * @callback: (scope async): callback to call when the request is satisfied.
+ * @user_data: (closure): the data to pass to callback function.
+ *
+ * Asynchronously check features of xapian bridge.
+ */
+void
+eknc_xapian_bridge_test (EkncXapianBridge *self,
+                         GCancellable *cancellable,
+                         GAsyncReadyCallback callback,
+                         gpointer user_data)
+{
+  g_return_if_fail (EKNC_IS_XAPIAN_BRIDGE (self));
+  g_return_if_fail (G_IS_CANCELLABLE (cancellable) || cancellable == NULL);
+
+  GTask *task = g_task_new (self, cancellable, callback, user_data);
+
+  RequestState *state = g_slice_new0 (RequestState);
+  g_task_set_task_data (task, state, request_state_free);
+
+  g_autoptr(SoupURI) uri = get_xapian_test_uri (self);
+  send_json_ld_request (self, uri, task);
+}
+
+/**
+ * eknc_xapian_bridge_test_finish:
+ * @self: the xapian bridge object
+ * @result: the #GAsyncResult that was provided to the callback.
+ *
+ * Finish a eknc_xapian_bridge_test call.
+ */
+void
+eknc_xapian_bridge_test_finish (EkncXapianBridge *self,
```

Comment:
I think it would be appropriate for this function to set an error and return `FALSE` when it can't talk to xapian-bridge, or there's an error during the communication. That of course would leave `has_default_op` set to `FALSE`, so it would be handled as an old-style xapian-bridge from the caller.